Core/Foundation/Source/DebugTrace.cpp

```diff
@@ -0,0 +1,45 @@
+#include <stdarg.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <functional>
+
+namespace Babylon
+{
+    namespace
+    {
+        static bool s_debugTraceEnabled{};
+        static std::function<void(const char* output)> s_traceOutputFunction;
+        static void TraceArgs(const char* format, va_list args)
+        {
+            char temp[8192];
+            auto len = vsnprintf(temp, sizeof(temp), format, args);
+            temp[len] = '\0';
+            s_traceOutputFunction(temp);
+        }
```

Comment:
Anonymous namespace doesn't need to be under the Babylon namespace.

Also, `static` is redundant inside an anonymous namespace.